File: src/main/java/meetme/users/UserService.java
package meetme.users;

import java.io.File;
import java.io.IOException;
import java.util.List;
import java.util.Map;

public interface UserService {
    void save(User user);
    User findById(String id);
    List<User> findAll();
    public Map uploadToCloudinary(File target) throws IOException;
}

File: src/main/java/meetme/users/UserService.kt
package meetme.users

import org.springframework.data.redis.core.PartialUpdate
import java.io.File
import java.io.IOException

interface UserService {
    fun save(user: User)

    fun findById(id: String): User?

    fun findAll(): List<User>

    @Throws(IOException::class)
    fun uploadToCloudinary(target: File): Map<*, *>

    fun findAllUserInterests(user: String): List<Pair<String, List<UserInterest>>>

    fun saveInterest(userInterest: List<UserInterest>, userId: String): PartialUpdate<User>
    fun updateUser(user: User, columnName: String? = null): PartialUpdate<User>
}
